tests/main/interfaces-mount-control/task.yaml

```diff
@@ -0,0 +1,73 @@
+summary: Test for the mount-control interface
+
+environment:
+    MOUNT_SRC: /var/tmp/test-snapd-mount-control
+    SNAP_COMMON: /var/snap/test-snapd-mount-control/common
+    SNAP_NAME: test-snapd-mount-control
+
+prepare: |
+    mkdir -p "$MOUNT_SRC/dir1"
+    echo "Something" > "$MOUNT_SRC/file1"
+
+restore: |
+    rm connect_error.log
+    rm -rf "$MOUNT_SRC"
+
+execute: |
+    MOUNT_DEST="$SNAP_COMMON/target"
+
+    echo "First verify that a snap with a malicious manifest cannot be connected"
+    "$TESTSTOOLS"/snaps-state install-local test-mount-control-invalid
+    snap connect test-mount-control-invalid:mntctl 2> connect_error.log || true
+    if os.query is-trusty; then
+        echo "On Trusty, we should fail anyway due to systemd being too old"
+        MATCH "systemd version 204 is too old" < connect_error.log
+        exit 0
+    fi
+
+    MATCH 'mount-control "where" pattern is not clean' < connect_error.log
+
+    echo "Installing the test snap"
+
+    "$TESTSTOOLS"/snaps-state install-local "${SNAP_NAME}"
+
+    echo "Connecting the mount-control interface"
+    snap connect "${SNAP_NAME}":mntctl
+
+    echo "Verify that the snap can perform a mount"
+    mkdir -p "$MOUNT_DEST"
+    "${SNAP_NAME}".cmd mount -o bind,rw "$MOUNT_SRC" "$MOUNT_DEST"
+
+    echo "Verify that the mount has been performed"
+    "${SNAP_NAME}".cmd grep "$MOUNT_DEST" /proc/self/mountinfo
+
+    echo "Ensure that the mounted files are visible"
+    "${SNAP_NAME}".cmd test -e "$MOUNT_DEST/file1"
+
+    echo "Unmount via the system command umount(8)"
+    "${SNAP_NAME}".cmd umount "$MOUNT_DEST"
+    if "${SNAP_NAME}".cmd grep "$MOUNT_DEST" /proc/self/mountinfo; then
+        echo "Unmount failed"
+        exit 1
+    fi
+    "${SNAP_NAME}".cmd test "!" -e "$MOUNT_DEST/file1"
+
+    echo "Verify that a mount not matching the allowed pattern will fail"
+    if "${SNAP_NAME}".cmd mount -o bind,rw "$MOUNT_SRC" "/tmp/"; then
+        echo "Mount succeeded despite not matching the allowed pattern"
+        if os.query is-opensuse && ! os.query is-opensuse-tumbleweed; then
```

Comment:
you could also just have a check which exits the test if `os.query is-opensuse-tumbleweed` is true before doing these checks, it would make the nested if's here a bit more readable/less indented

Reply:
Indeed; updated.